### DISCUSSION TITLE
[Error] importlib.metadata.PackageNotFoundError :qtile

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [qtile/qtile#3090] [Q&A] [answered]
`importlib.metadata.PackageNotFoundError`
I can't start qtile or use it at all, not even `qtile --version`...It happened right after i updated to the latest version of qtile.
![PXL_20211213_105753485 NIGHT](https://user-images.githubusercontent.com/61395246/145783509-5a8a7b84-bee4-4e72-9281-3533d068aac9.jpg)
Forgive me for the self phone picture but don't know any other way to share stuff from the tty.

I tried opening an issue but it kept giving me an error, en because this is really important to me i just rushed to the discussion.
Any help would be appreciated!

**Comment by elParaguayo** [comment]
Have you recently upgraded to python 3.10 on your system? I'm just asking because I think it's available in the Arch repos now but your error message above is referring to python 3.9.

**Reply by LamprosPitsillos** [reply]
`python --version` gives me 3.9.9

**Reply by elParaguayo** [reply]
OK. So it's not that.

How did you install qtile?

**Reply by LamprosPitsillos** [reply]
Via pacman, on Artix Linux

**Comment by LamprosPitsillos** [comment]
Running `rg "upgraded" /var/log/pacman.log` i see that i upgrade also this packages
![PXL_20211213_120529971 NIGHT](https://user-images.githubusercontent.com/61395246/145792470-c77fac00-6a51-49f3-aa5b-702852dd4533.jpg)
Again...sorry for the self phone pic.

**Reply by elParaguayo** [reply]
Hmm. Arch have repackaged qtile. Not sure why they've gone to 1-4. Maybe that's related to python 3.10 so it would force a reinstall.

**Reply by LamprosPitsillos** [reply]
What can i do about this?

**Reply by elParaguayo** [reply]
Can you take a look at `pacman -Ql qtile` and check that the libraries are installed in `/usr/lib/python3.9` (don't need to paste output here)

**Comment by LamprosPitsillos** [comment]
Tried down grading to 1-3..no luck there either...
![PXL_20211213_122816139 NIGHT](https://user-images.githubusercontent.com/61395246/145796025-cb2a279c-7d57-4f16-b478-4cf3cc030355.jpg)

**Reply by elParaguayo** [reply]
1-4 was just a version bump to force a reinstall on python 3.10 so I'm not sure why you've got that package but not 3.10.

**Reply by LamprosPitsillos** [reply]
Also my terminal ,Kitty is broken... `libpython3.10.so.1.0 no such file`

**Comment by LamprosPitsillos** [comment]
From what i understand, i am at the mercy of the Artix repo maintainers...

**Reply by elParaguayo** [reply]
Yeah - looks like you need python 3.10.

**Comment by LamprosPitsillos** [comment]
![PXL_20211213_140929017 NIGHT](https://user-images.githubusercontent.com/61395246/145810223-4cb597f1-8ae6-46ff-8b47-2fdb22ee24dd.jpg)
Upgraded manually to python 10.
Errors still appear, and now even pip doesn't work...

**Reply by elParaguayo** [reply]
When a distro does an upgrade of python, every python module also needs to be upgraded at the same time. It looks like you still have other modules that have not been installed for python 3.10 (e.g. you probably haven't reinstalled pip).

For example, if you look at this page: https://archlinux.org/packages/?sort=&q=python-&maintainer=&flagged=  you'll see that all the python packages have the same update date so that they're all reinstalled alongside python 3.10.

I haven't yet upgraded to 3.10 but will probably do it soon.

**Reply by elParaguayo** [reply]
Also, when you say you installed python 3.10 "manually", how did you do it? I wouldn't recommend installing packages outside of pacman.

**Reply by LamprosPitsillos** [reply]
Pacman -U *wget the python 10 from the arch repo*

**Comment by LamprosPitsillos** [comment, marked as answer]
Managed to downgrade everything from the latest update. Now everything is back to normal. I will close this since it's not qtiles fault. Thank you very much for your time!

**Reply by elParaguayo** [reply]
No problem.

Bit nervous about my upgrade to python 3.10 now!

**Reply by LamprosPitsillos** [reply]
I wouldn't worry , i will update again as soon as i see that the python package gets updated as well.
Good luck!

**Reply by m-col** [reply]
I updated to Python 3.10 today too (on Arch) and had no problems. Seems like something has gone wrong either as part of your update or system or in the way Artix has handled the update.

**Reply by LamprosPitsillos** [reply]
Most likely the Artix repos fault. It pushed the updated packages without even pushing python 3.10.

**Reply by m-col** [reply]
Ah I see, what a mistake!
